graylog2-web-interface/src/components/pipelines/StageForm.jsx

```diff
@@ -71,24 +72,37 @@ const StageForm = createReactClass({
 
   _saved() {
     this._closeModal();
-    if (this.props.create) {
+    const { create } = this.props;
+    if (create) {
       this.setState(this.getInitialState());
     }
   },
 
   _save() {
-    this.props.save(this.state.stage, this._saved);
+    const { stage } = this.state;
+    const { save } = this.props;
+    save(stage, this._saved);
   },
 
   _getFormattedOptions(rules) {
-    return rules ? rules.map((rule) => {
-      return { value: rule.title, label: rule.title };
-    }) : [];
+    const { stage } = this.state;
+    const choosedRules = stage.rules;
+    return rules ? rules.filter(rule => this._filterChoosedRules(rule, choosedRules)).map(this._formatRuleOption) : [];
+  },
+
+  _formatRuleOption(rule) {
+    return { value: rule.title, label: rule.title };
+  },
+
+  _filterChoosedRules(rule, choosedRules) {
```

Comment:
Small nitpick: we should use the right tense here
```suggestion
  _filterChosenRules(rule, chosenRules) {
```